### PR TITLE
[WIP] Replace with prost

### PR DESCRIPTION
**Pull request by ice1000**
TODO:

+ [ ] Deal with [`raft::eraftpb`](https://docs.rs/raft/0.4.0/raft/eraftpb/struct.Entry.html)

Please don't merge until #347 is merged and I've rebased this PR onto that one.

### CODE REVIEW

**Comment by ice1000** [comment]
> Looks like grpc code is removed?

Yes. Do you mean we shouldn't?

**Comment by ice1000** [comment]
Sounds like we shouldn't. Reverted.

**Comment by ice1000** [comment]
Let's wait for #347 for a solution to `*_grpc.rs`-related generation.

File: build.rs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    file.sync_all()?;
+    Ok((import_all_path, protos))
+}
+
+fn main() {
+    let (import_all_path, protos) = generate_import_all_proto().unwrap();
+
+    prost_build::compile_protos(
+        &["proto/import_all.proto"],
+        &["/home/ice1000/git-repos/PingCAP/kvproto/proto",
```
Comment (siddontang) [inline build.rs]: don't use your absoluate path here

Comment (ice1000) [reply build.rs]: Oh! I'm sorry. This should be a forget-to-change. I use this for testing and it's supposed to be something like `current_dir, current_dir+"/proto", current_dir+"/include"`

**Comment by ice1000** [comment]
Closed in favor of #360

**Pull request closed without merging**